### PR TITLE
Add retry for stream creation

### PR DESCRIPTION
**Pull request by SGauvin**
It is possible for stream creation to fail (I have seen it twice).
To try to mitigate this issue, we retry up to two times to create the audio stream with exponential time delays.

### CODE REVIEW

File: src/audio.rs [hunk 1]
```diff
@@ -75,16 +77,21 @@ fn start_stream(
     config: &StreamConfig,
     audio_device: &Device,
     sample_format: &SampleFormat,
-) -> (cpal::Stream, Receiver<i16>) {
-    let (tx, rx) = sync::mpsc::channel();
+) -> Result<(cpal::Stream, Receiver<i16>), cpal::BuildStreamError> {
+    println!("Starting audio stream");
```
Comment (samichoulo911) [inline src/audio.rs]: Shouldn't this be "Attempting to start audio stream" ?

File: src/audio.rs [hunk 2]
```diff
@@ -75,16 +77,21 @@ fn start_stream(
...
         SampleFormat::I16 => build_audio_stream::<i16>(audio_device, config, tx),
         SampleFormat::F32 => build_audio_stream::<f32>(audio_device, config, tx),
-    }
-    .expect("Failed to create audio stream");
-    // TODO handle error
+    };
 
-    (stream, rx)
+    match stream {
+        Ok(stream) => Ok((stream, rx)),
```
Comment (samichoulo911) [inline src/audio.rs]: Here add "Audio stream started"

Comment (SGauvin) [reply src/audio.rs]: Je crois pas qu'on devrait avoir 2 logs pour kkch d'aussi simple que de start le stream audio

Comment (SGauvin) [reply src/audio.rs]: @samichoulo911 jai modifié comment je log lmk si cest good

**Review by samichoulo911 — APPROVED** [review]
LGTM

**Pull request merged**